src/dmd/dmodule.d

```diff
@@ -662,7 +662,27 @@ extern (C++) final class Module : Package
             return true; // already read
 
         //printf("Module::read('%s') file '%s'\n", toChars(), srcfile.toChars());
-        if (auto result = global.fileManager.lookup(srcfile))
+
+        /* Preprocess the file if it's a C file
+         */
+        FileName filename = srcfile;
+        bool ifile = false;             // did we generate a .i file
+        scope (exit)
+        {
+            if (ifile)
+                File.remove(filename.toChars());        // remove generated file
+        }
+        version (posix)
```

Comment:
Posix, also it'd be good to move this version condition to dmd.mars where you set preprocess. As there is no reason why gdc (and ldc?) can't do this on Windows.

Reply:
I don't really know what it will take to make things work for gdc and ldc. My focus here is getting *something* to work.

Reply:
It's a step towards that. This is mixing driver/implementation-specific logic in `dmodule.d`, which should be generic to any handler and there is nothing that prevents it to run the preprocessor on other platforms.